### PR TITLE
Use rustfmt to hint at clean ups for the codebase

### PR DESCRIPTION
**Pull request by tcharding**
`rustfmt` is still under discussion, while researching the topic I came across a maintainer of another project that does not use `rustfmt` who mentioned that he manually implemented the `rusfmt` suggestions that he liked ever month or so. This seemed like a good idea so I did it. This was extremely painful but I believe I have put together a PR that is non-controversial with well separated patches.

Totally non urgent.

### CODE REVIEW

File: src/blockdata/script.rs [hunk 1]
```diff
@@ -239,9 +238,9 @@ pub fn read_scriptint(v: &[u8]) -> Result<i64, Error> {
 /// else as true", except that the overflow rules don't apply.
 #[inline]
 pub fn read_scriptbool(v: &[u8]) -> bool {
-    !(v.is_empty() ||
-        ((v[v.len() - 1] == 0 || v[v.len() - 1] == 0x80) &&
-         v.iter().rev().skip(1).all(|&w| w == 0)))
+    !(v.is_empty()
+      || ((v[v.len() - 1] == 0 || v[v.len() - 1] == 0x80)
+          && v.iter().rev().skip(1).all(|&w| w == 0)))
```
Comment (Kixunil) [inline src/blockdata/script.rs:243]: Oh, at first I thought this is a closure. This whole chunk is super-hard to read. I believe the proper solution for such cases is to assign the values to clearly named variables and then have an inline expression (`!v.is_empty() || (var_a && car_b))

However note that `is_empty()` is actually bounds check for subsequent operations which makes it even uglier. This should help:
```rust
let last = math v.last() {
    Some(last) => last,
    None => return false,
};
// use `last` from now on
```

File: src/blockdata/script.rs [hunk 2]
```diff
@@ -417,32 +416,32 @@ impl Script {
...
-        self.0[1] == opcodes::all::OP_HASH160.into_u8() &&
-        self.0[2] == opcodes::all::OP_PUSHBYTES_20.into_u8() &&
-        self.0[23] == opcodes::all::OP_EQUALVERIFY.into_u8() &&
-        self.0[24] == opcodes::all::OP_CHECKSIG.into_u8()
+        self.0.len() == 25
+            && self.0[0] == opcodes::all::OP_DUP.into_u8()
+            && self.0[1] == opcodes::all::OP_HASH160.into_u8()
+            && self.0[2] == opcodes::all::OP_PUSHBYTES_20.into_u8()
+            && self.0[23] == opcodes::all::OP_EQUALVERIFY.into_u8()
+            && self.0[24] == opcodes::all::OP_CHECKSIG.into_u8()
```
Comment (Kixunil) [inline src/blockdata/script.rs:433]: Being uniform, these are not as bad as the one above.

File: src/blockdata/script.rs [hunk 3]
```diff
@@ -417,32 +416,32 @@ impl Script {
...
-            self.0[66] == opcodes::all::OP_CHECKSIG.into_u8())
-     || (self.0.len() == 35 &&
-            self.0[0] == opcodes::all::OP_PUSHBYTES_33.into_u8() &&
-            self.0[34] == opcodes::all::OP_CHECKSIG.into_u8())
+        (self.0.len() == 67
+         && self.0[0] == opcodes::all::OP_PUSHBYTES_65.into_u8()
+         && self.0[66] == opcodes::all::OP_CHECKSIG.into_u8())
+            || (self.0.len() == 35
+                && self.0[0] == opcodes::all::OP_PUSHBYTES_33.into_u8()
+                && self.0[34] == opcodes::all::OP_CHECKSIG.into_u8())
```
Comment (Kixunil) [inline src/blockdata/script.rs:444]: `match self.0.len()` would be nice here.

**Review by Kixunil — COMMENTED** [review]
This reminded me that some formatting issues can not be solved by automated tool. At best there could be an automated tool that complains about them.

**Review by apoelstra — APPROVED** [review]
ACK fa4da46e229d386b0fedf039b8f177cb3ff96e5a

None of these seem problematic to me. Thanks for all the effort filtering and categorizing these!

**Comment by tcharding** [comment]
I removed the patch that does logical operators and force pushed, that means commint hash of the last 4 patches has change, no other changes introduced, please re-ack @apoelstra.

I'll have a play with the logical operators as a separate PR. Thanks for the reviews!

File: src/blockdata/script.rs (hunk 1 above)
Comment (tcharding) [reply src/blockdata/script.rs]: Ha ha, its funny I didn't see that as a closure, soon as I started refactoring this code I dropped straight back into being a C programmer. The human brain is a mysterious thing.

Comment (tcharding) [reply src/blockdata/script.rs]: Please see: https://github.com/rust-bitcoin/rust-bitcoin/pull/808

**Comment by tcharding** [comment]
Please see: https://github.com/rust-bitcoin/rust-bitcoin/pull/808

**Review by apoelstra — DISMISSED** [review]
ACK e6aa68bf1d5492f1ca106381b290eef10e24649f

**Comment by tcharding** [comment]
The change in https://github.com/rust-bitcoin/rust-bitcoin/pull/806/commits/011618dcef1c891e18acd0ee92622cd5e287f78a to the file `blockdata/transaction.rs` is slightly incorrect. The opening brace should be on a newline. So as not to make @apoelstra have to re-review the whole set I'm going to go against usual process and add an additional patch to the end of this PR.

I also missed doing the `where` statement on `verify_with_flags` in the same file :(

**Review by apoelstra — DISMISSED** [review]
ACK 13b451f79294139ce255082f8921a01ea3134664

FWIW I don't mind if you edit old commits. I can use `range-diff` to see the edits.

**Comment by dr-orlovsky** [comment]
Now requires rebase due to doc changes merged :(

**Comment by tcharding** [comment]
Changes in force push:
- Squash 'fix where statements patches'
- Pull logic change out of whitespace only patch (added final patch 'remove unnecessary type annotation') 
- Rebase on master

**Review by apoelstra — APPROVED** [review]
ACK a77907d59c00d2c879461e41b537227696fc610b

**Review by sanket1729 — APPROVED** [review]
ACK a77907d59c00d2c879461e41b537227696fc610b. 

I ran `cargo fmt` before the first commit and at the tip of this PR and noted that the diff was non-logical changes only. 

@tcharding , some things that might also be included like spaces in `bip158.rs`, but that can be done in a later PR. 

I think we should merge this PR now that it has two ACKs because I can imagine this will conflict with multiple things and will be annoying to maintain. Plus, it is a non-logical change so can get it before RC too. 

```diff
diff --git a/src/blockdata/script.rs b/src/blockdata/script.rs
index 3e4d4f9..6f9dbc0 100644
--- a/src/blockdata/script.rs
+++ b/src/blockdata/script.rs
@@ -1484,14 +1484,14 @@ mod test {
         let slop_v_nonmin_alt: Result<Vec<Instruction>, Error> =
             nonminimal_alt.instructions().collect();
 
-        assert_eq!(v_zero.unwrap(), vec![Instruction::PushBytes(&[])]);
-        assert_eq!(v_zeropush.unwrap(), vec![Instruction::PushBytes(&[0])]);
+        assert_eq!(v_zero.unwrap(), vec![Instruction::PushBytes(&[]),]);
+        assert_eq!(v_zeropush.unwrap(), vec![Instruction::PushBytes(&[0]),]);
 
         assert_eq!(
             v_min.clone().unwrap(),
             vec![
                 Instruction::PushBytes(&[105]),
-                Instruction::Op(opcodes::OP_NOP3)
+                Instruction::Op(opcodes::OP_NOP3),
             ]
         );
 
@@ -1501,7 +1501,7 @@ mod test {
             v_nonmin_alt.clone().unwrap(),
             vec![
                 Instruction::PushBytes(&[105, 0]),
-                Instruction::Op(opcodes::OP_NOP3)
+                Instruction::Op(opcodes::OP_NOP3),
             ]
         );
 
diff --git a/src/blockdata/transaction.rs b/src/blockdata/transaction.rs
index 311fd08..b1fabfb 100644
--- a/src/blockdata/transaction.rs
+++ b/src/blockdata/transaction.rs
@@ -723,7 +723,7 @@ impl fmt::Display for NonStandardSigHashType {
 #[cfg_attr(docsrs, doc(cfg(feature = "std")))]
 impl error::Error for NonStandardSigHashType {}
 
-/// Legacy Hashtype of an input's signature
+/// Legacy Hashtype of an input's signature.
 #[deprecated(since = "0.28.0", note = "Please use [`EcdsaSigHashType`] instead")]
 pub type SigHashType = EcdsaSigHashType;
 
diff --git a/src/blockdata/witness.rs b/src/blockdata/witness.rs
index d159d37..645a33d 100644
--- a/src/blockdata/witness.rs
+++ b/src/blockdata/witness.rs
@@ -366,7 +366,6 @@ mod test {
             expected_wit[0],
             tx.input[0].witness.second_to_last().unwrap().to_hex()
         );
-
         let tx_bytes_back = serialize(&tx);
         assert_eq!(tx_bytes_back, tx_bytes);
     }
diff --git a/src/consensus/encode.rs b/src/consensus/encode.rs
index b97c0f1..caf01c4 100644
--- a/src/consensus/encode.rs
+++ b/src/consensus/encode.rs
@@ -734,7 +734,7 @@ impl<T: Encodable> Encodable for sync::Arc<T> {
 
 // Tuples
 macro_rules! tuple_encode {
-    ($($x:ident),*) => {
+    ($($x:ident),*) => (
         impl <$($x: Encodable),*> Encodable for ($($x),*) {
             #[inline]
             #[allow(non_snake_case)]
@@ -756,7 +756,7 @@ macro_rules! tuple_encode {
                 Ok(($({let $x = Decodable::consensus_decode(&mut d)?; $x }),*))
             }
         }
-    };
+    );
 }
 
 tuple_encode!(T0, T1);
diff --git a/src/network/stream_reader.rs b/src/network/stream_reader.rs
index 1cf4a9e..cf6e64c 100644
--- a/src/network/stream_reader.rs
+++ b/src/network/stream_reader.rs
@@ -213,7 +213,7 @@ mod test {
         let istream = TcpStream::connect(format!("127.0.0.1:{}", port)).unwrap();
         let reader = BufReader::new(istream);
 
-        (handle, reader)
+        return (handle, reader);
     }
 
     #[test]
diff --git a/src/util/address.rs b/src/util/address.rs
index b3c3e95..6eda874 100644
--- a/src/util/address.rs
+++ b/src/util/address.rs
@@ -98,10 +98,17 @@ impl fmt::Display for Error {
             Error::InvalidWitnessVersion(v) => write!(f, "invalid witness script version: {}", v),
             Error::UnparsableWitnessVersion(_) => write!(f, "incorrect format of a witness version byte"),
             Error::MalformedWitnessVersion => f.write_str("bitcoin script opcode does not match any known witness version, the script is malformed"),
-            Error::InvalidWitnessProgramLength(l) => write!(f, "the witness program must be between 2 and 40 bytes in length: length={}", l),
-            Error::InvalidSegwitV0ProgramLength(l) => write!(f, "a v0 witness program must be either of length 20 or 32 bytes: length={}", l),
-            Error::UncompressedPubkey => write!(f, "an uncompressed pubkey was used where it is not allowed"),
-            Error::ExcessiveScriptSize => write!(f, "Script size exceed 520 bytes"),
+            Error::InvalidWitnessProgramLength(l) => write!(f,
+                "the witness program must be between 2 and 40 bytes in length: length={}", l,
+            ),
+            Error::InvalidSegwitV0ProgramLength(l) => write!(f,
+                "a v0 witness program must be either of length 20 or 32 bytes: length={}", l,
+            ),
+            Error::UncompressedPubkey => write!(f,
+                "an uncompressed pubkey was used where it is not allowed",
+            ),
+            Error::ExcessiveScriptSize => write!(f,
+                "Script size exceed 520 bytes")
         }
     }
 }
diff --git a/src/util/bip158.rs b/src/util/bip158.rs
index 10ee4e9..f96d532 100644
--- a/src/util/bip158.rs
+++ b/src/util/bip158.rs
@@ -29,21 +29,21 @@
 //! fn get_script_for_coin(coin: &OutPoint) -> Result<Script, BlockFilterError> {
 //!   // get utxo ...
 //! }
-//!
+//!  
 //! // create a block filter for a block (server side)
 //! let filter = BlockFilter::new_script_filter(&block, get_script_for_coin)?;
 //!
 //! // or create a filter from known raw data
 //! let filter = BlockFilter::new(content);
-//!
+//!  
 //! // read and evaluate a filter
-//!
+//!  
 //! let query: Iterator<Item=Script> = // .. some scripts you care about
 //! if filter.match_any(&block_hash, &mut query.map(|s| s.as_bytes())) {
 //!   // get this block
 //! }
 //!  ```
-//!
+//!  
 
 use prelude::*;
 
diff --git a/src/util/key.rs b/src/util/key.rs
index 81d6412..706237e 100644
--- a/src/util/key.rs
+++ b/src/util/key.rs
@@ -187,7 +187,7 @@ impl PublicKey {
 
     /// Deserialize a public key from a slice
     pub fn from_slice(data: &[u8]) -> Result<PublicKey, Error> {
-        let compressed = match data.len() {
+        let compressed: bool = match data.len() {
             33 => true,
             65 => false,
             len => {
diff --git a/src/util/psbt/mod.rs b/src/util/psbt/mod.rs
index 0bae3db..4f8ebc5 100644
--- a/src/util/psbt/mod.rs
+++ b/src/util/psbt/mod.rs
@@ -902,7 +902,7 @@ mod tests {
             assert_eq!(
                 tx.txid(),
                 Txid::from_hex("75c5c9665a570569ad77dd1279e6fd4628a093c4dcbf8d41532614044c14c115")
-                    .unwrap(),
+                    .unwrap()
             );
 
             let mut unknown: BTreeMap<raw::Key, Vec<u8>> = BTreeMap::new();
diff --git a/src/util/taproot.rs b/src/util/taproot.rs
index 1bf3a7f..7546da9 100644
--- a/src/util/taproot.rs
+++ b/src/util/taproot.rs
@@ -987,13 +987,11 @@ impl fmt::Display for TaprootBuilderError {
                 "Attempted to create a tree with two nodes at depth 0. There must\
                 only be a exactly one node at depth 0",
             ),
-            TaprootBuilderError::InvalidMerkleTreeDepth(d) => {
-                write!(
-                    f,
-                    "Merkle Tree depth({}) must be less than {}",
-                    d, TAPROOT_CONTROL_MAX_NODE_COUNT
-                )
-            }
+            TaprootBuilderError::InvalidMerkleTreeDepth(d) => write!(
+                f,
+                "Merkle Tree depth({}) must be less than {}",
+                d, TAPROOT_CONTROL_MAX_NODE_COUNT
+            ),
             TaprootBuilderError::InvalidInternalKey(e) => {
                 write!(f, "Invalid Internal XOnly key : {}", e)
             }
```

**Comment by tcharding** [comment]
Thanks @sanket1729, there are definitely more things that could be done. I just did a subset that I thought was non-controversial and did not do too much code churn (e.g. adding commas to lines, I actually started doing them then dropped it :)

**Comment by apoelstra** [comment]
I'm gonna merge this. Hopefully it does not break too many PRs. I think we have only one -- #796 -- that we need to get in before the RC, in any case, and it has 3 ACKs (but some outstanding concerns from Kixunil).

**Pull request merged**